### PR TITLE
feat: add stats

### PR DESCRIPTION
**Pull request by abulte**
Fix https://github.com/ecolabdata/ecospheres/issues/499

```
dashboard_backend=# select * from stats where date = '2024-11-21';
-[ RECORD 1 ]--------------+-----------
id                         | 235
date                       | 2024-11-21
nb_uniq_visitors           | 583
nb_visits                  | 641
nb_actions                 | 4388
nb_visits_converted        | 0
bounce_count               | 226
sum_visit_length           | 154586
max_actions                | 107
bounce_rate                | 0.35
nb_actions_per_visit       | 6.8
avg_time_on_site           | 241
nb_pageviews               | 2600
nb_downloads               | 106
nb_uniq_visitors_returning | 67
nb_uniq_visitors_new       | 552
```

### CODE REVIEW

File: alembic/versions/ccdd23b19763_add_stats_table.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+from typing import Sequence, Union
+
+import sqlalchemy as sa
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "ccdd23b19763"
+# FIXME: check that https://github.com/ecolabdata/ecospheres-dashboard-backend/pull/29 has not been merged first
```
Comment (abulte) [inline alembic/versions/ccdd23b19763_add_stats_table.py:17]: FIXME

Comment (abulte) [reply alembic/versions/ccdd23b19763_add_stats_table.py:17]: Protip : `ALEMBIC_ENV=demo alembic history -r head:head` on `main`

**Review by DanFluture275 — APPROVED** [review]

File: cli.py [hunk 2]
```diff
@@ -249,6 +254,69 @@ def add_metric(
...
+    # defaults to yesterday
+    parsed_day = date.fromisoformat(day) if day else date.today() - timedelta(days=1)
+    print(f"Loading stats for {parsed_day.isoformat()}...")
+
+    stats_url = get_config_value(env, "stats_url")
+    stats_site_id = get_config_value(env, "stats_site_id")
+    stats_token = get_config_value(env, "stats_token")
+
+    if not stats_url or not stats_site_id or not stats_token:
+        print("Skipping stats loading: missing config value(s)")
```
Comment (streino) [inline cli.py:281]: Might be worth a `log.warn`

File: cli.py [hunk 3]
```diff
@@ -249,6 +254,69 @@ def add_metric(
...
+    methods = ["VisitsSummary.get", "Actions.get", "VisitFrequency.get"]
+    for method in methods:
+        data |= fetch(method)
+
+    columns = [column.key for column in inspect(Stats).attrs if column.key != "id"]
+    db_data = {k: v for k, v in data.items() if k in columns}
+    db_data["date"] = parsed_day
+    if "bounce_rate" in db_data:
+        # 39% -> 0.39
+        db_data["bounce_rate"] = float(db_data["bounce_rate"].replace("%", "")) / 100
```
Comment (streino) [inline cli.py:314]:
```suggestion
        db_data["bounce_rate"] = float(db_data["bounce_rate"].rstrip("%")) / 100
```

File: config.py [hunk 4]
```diff
@@ -10,12 +12,18 @@
...
+        "stats_url": "https://stats.data.gouv.fr/index.php",
+        "stats_site_id": "299",
+        "stats_token": os.getenv("STATS_TOKEN", ""),
     },
     "demo": {
         "universe_name": "ecospheres",
         "topic_slug": "univers-ecospheres",
         "prefix": "demo",
         "dsn": os.getenv("DATABASE_URL", ""),
+        "stats_url": "",
```
Comment (streino) [inline config.py:24]: Better as `None`? (here and below)

Comment (abulte) [reply config.py:24]: Yeah, but I would have to handle None values for every conf key, or precisely type which can be None... That's why I settled for this hack.

File: cli.py (hunk 2 above)
Comment (abulte) [reply cli.py:281]: Ideally yes, but we don't have a logger yet. I'd keep this as is for now (not an error), and add a logger later.

**Review by streino — APPROVED** [review]

**Pull request merged**